Tasks/AzureFunctionDeploymentV1/deploymentProvider/WindowsWebAppZipDeployProvider.ts

```diff
@@ -0,0 +1,58 @@
+import { AzureRmWebAppDeploymentProvider } from './AzureRmWebAppDeploymentProvider';
+import tl = require('vsts-task-lib/task');
+import * as ParameterParser from 'azurermdeploycommon/operations/ParameterParserUtility'
+import { DeploymentType } from '../taskparameters';
+import { PackageType } from 'azurermdeploycommon/webdeployment-common/packageUtility';
+import { FileTransformsUtility } from 'azurermdeploycommon/operations/FileTransformsUtility.js';
+const deleteOldRunFromZipAppSetting: string = '-WEBSITE_RUN_FROM_ZIP';
+const removeRunFromZipAppSetting: string = '-WEBSITE_RUN_FROM_PACKAGE 0';
+var deployUtility = require('azurermdeploycommon/webdeployment-common/utility.js');
+var zipUtility = require('azurermdeploycommon/webdeployment-common/ziputility.js');
+
+export class WindowsWebAppZipDeployProvider extends AzureRmWebAppDeploymentProvider{
+    
+    private zipDeploymentID: string;
+ 
+    public async DeployWebAppStep() {
+        var webPackage = await FileTransformsUtility.applyTransformations(this.taskParams.Package.getPath(), this.taskParams.WebConfigParameters, this.taskParams.Package.getPackageType());
+
+        if(this.taskParams.DeploymentType === DeploymentType.zipDeploy) {
+            var _isMSBuildPackage = await this.taskParams.Package.isMSBuildPackage();
+            if(_isMSBuildPackage) {
+                throw Error(tl.loc("Publishusingzipdeploynotsupportedformsbuildpackage"));
+            }
+            else if(this.taskParams.Package.getPackageType() === PackageType.war) {
+                throw Error(tl.loc("Publishusingzipdeploydoesnotsupportwarfile"));
+            }
+        }
+
+        if(tl.stats(webPackage).isDirectory()) {
+            let tempPackagePath = deployUtility.generateTemporaryFolderOrZipPath(tl.getVariable('AGENT.TEMPDIRECTORY'), false);
+            webPackage = await zipUtility.archiveFolder(webPackage, "", tempPackagePath);
+            tl.debug("Compressed folder into zip " +  webPackage);
+        }
+
+        tl.debug("Initiated deployment via kudu service for webapp package : ");
+        
+        var updateApplicationSetting = ParameterParser.parse(removeRunFromZipAppSetting)
```

Comment:
var updateApplicationSetting = ParameterParser.parse(removeRunFromZipAppSetting)
  var deleteApplicationSetting = ParameterParser.parse(deleteOldRunFromZipAppSetting)

Can we put this inside appServiceUtility (updateAndMonitorAppSettings) then all provider can reuse it and code will looks clear .